src/app/app-wrapper.js

```diff
@@ -1,21 +1,43 @@
 import { CssReset, CssVariables } from '@dhis2/ui'
 import React from 'react'
-import { QueryClientProvider } from 'react-query'
+import { ReactQueryDevtools } from 'react-query/dist/react-query-devtools.development'
+import { PersistQueryClientProvider } from 'react-query/persistQueryClient'
 import { HashRouter as Router, Route } from 'react-router-dom'
 import { QueryParamProvider } from 'use-query-params'
 import { RightHandPanelProvider } from '../right-hand-panel/index.js'
 import { CurrentItemProvider } from '../shared/index.js'
 import App from './app.js'
+import createIDBPersister from './query-client/persister.js'
 import useQueryClient from './query-client/use-query-client.js'
 
+const persister = createIDBPersister()
+
+const persistOptions = {
```

Comment:
Should we export this from `persistor.js`, and import here instead of specifying the options here?

Reply:
There's an issue in regards to the hook you've touched: [TECH-1124](https://dhis2.atlassian.net/browse/TECH-1124). It's scheduled to be worked on post-release, but now that you've touched this area, I think this could be a good time. I really don't like the name `useQueryClient` because it's conflicting with the hook with the same name exported by `react-query`, which is why I was suggesting a different name.

I can see why you've moved the persister stuff from the hook to the `app-wrapper.js`. To me it'd make sense to keep the react-query related code in a separate file entirely, e.g. in a `<ConfiguedQueryClientProvider />` component file that internally creates the client instance and configures it, regardless of the implementation detail (with a config passed to the provider or through a hook).

Reply:
Agreed,  I will move this setup to a `ConfiguedQueryClientProvider`-component.

I'm not sure I necessarily agree that the naming is that bad, though. And I wouldn't say that it's a "completely different thing". The hook sets up a `react-query` `queryClient`, and thus can be used as an "override" for `useQueryClient`, so I actually think the naming is fair. But yeah, it's conflicting and you could probably be more specific like `useConfiguredQueryClient`.

Reply:
I agree it would be good to rename "our" `useQueryClient` to `useConfiguredQueryClient` in the long run. We do import `useQueryClient` from 'react-query' in a few places as well, so it can be slightly confusing to use the same variable name for two distinct entities. But this can remain a task for after the soft freeze IMO.

Reply:
Let's handle the `useQueryClient`-thing in a separate PR.